core/src/mesh.cpp

```diff
@@ -0,0 +1,89 @@
+
+#include <knoting/log.h>
+#include <knoting/mesh.h>
+#include <filesystem>
+
+namespace knot {
+bgfx::VertexLayout VertexLayout::meshVertexLayout;
+}  // namespace knot
+
+namespace knot {
+
+Mesh::Mesh() {}
+
+Mesh::~Mesh() {
+    bgfx::destroy(m_vbh);
+    bgfx::destroy(m_ibh);
+}
+
+void knot::Mesh::load_mesh(const std::string& localTexturePath) {
+    std::string fullPath = PATH_MODELS + localTexturePath;
+    std::filesystem::path fs_path = std::filesystem::path(fullPath);
+    bgfx::TextureHandle handle = BGFX_INVALID_HANDLE;
+
+    uint16_t size;
+    char* raw_image;
```

Comment:
Maybe we can create our own Image type which can store this kind of information.